### PR TITLE
meteo-qt: init at 3.4

### PR DESCRIPTION
**Pull request by linuxissuper**
## Description of changes

add meteo-qt
>System tray application for weather status information   

from https://github.com/dglent/meteo-qt

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/meteo-qt/package.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    ];
+
+  propagatedBuildInputs = with python3Packages; [
+    lxml
+    pyqt5
+  ];
+
+  pythonImportsCheck = [ "meteo_qt" ];
+
+  makeWrapperArgs = [ "\${qtWrapperArgs[@]}" ];
```
Comment (Aleksanaa) [inline pkgs/by-name/me/meteo-qt/package.nix:36]: Also needs `dontWrapQtApps = true`?

File: pkgs/by-name/me/meteo-qt/package.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  meta = with lib; {
+    description = "System tray application for weather status information";
+    homepage = "https://github.com/dglent/meteo-qt";
+    changelog = "https://github.com/dglent/meteo-qt/blob/${src.rev}/CHANGELOG";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ linuxissuper ];
```
Comment (Aleksanaa) [inline pkgs/by-name/me/meteo-qt/package.nix:43]:
```suggestion
  meta = {
    description = "System tray application for weather status information";
    homepage = "https://github.com/dglent/meteo-qt";
    changelog = "https://github.com/dglent/meteo-qt/blob/${src.rev}/CHANGELOG";
    license = lib.licenses.gpl3Only;
    maintainers = with lib.maintainers; [ linuxissuper ];
```

File: pkgs/by-name/me/meteo-qt/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    ]
+    ++ [
+      qt5.qttools
+      qt5.wrapQtAppsHook
+    ];
+
+  propagatedBuildInputs = with python3Packages; [
+    lxml
+    pyqt5
+  ];
```
Comment (Aleksanaa) [inline pkgs/by-name/me/meteo-qt/package.nix:32]:
```suggestion
  nativeBuildInputs = [
      qt5.qttools
      qt5.wrapQtAppsHook
    ];

  build-system = with python3Packages; [ sip ];

  dependencies = with python3Packages; [
    lxml
    pyqt5
  ];
```

File: pkgs/by-name/me/meteo-qt/package.nix (hunk 1 above)
Comment (linuxissuper) [reply pkgs/by-name/me/meteo-qt/package.nix:36]: seems to make no difference? 🤷   
but without `makeWrapperArgs` I get:  
`qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""`

File: pkgs/by-name/me/meteo-qt/package.nix (hunk 2 above)
Comment (linuxissuper) [reply pkgs/by-name/me/meteo-qt/package.nix:40]: I don't have a problem with this suggestion but why is it better?

File: pkgs/by-name/me/meteo-qt/package.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/by-name/me/meteo-qt/package.nix:33]: > seems to make no difference? 🤷

Find if there's `./result/bin/.meteo-qt_wrapped_wrapped`

Comment (linuxissuper) [reply pkgs/by-name/me/meteo-qt/package.nix:33]: only `.meteo-qt-wrapped` is there
regardless of `dontWrapQtApps = true;`

File: pkgs/by-name/me/meteo-qt/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "meteo-qt";
+  version = "3.4";
+
+  src = fetchFromGitHub {
+    owner = "dglent";
+    repo = "meteo-qt";
+    rev = "v${version}";
```
Comment (FliegendeWurst) [inline pkgs/by-name/me/meteo-qt/package.nix:15]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/me/meteo-qt/package.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  qt5,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "meteo-qt";
+  version = "3.4";
+
```
Comment (FliegendeWurst) [inline pkgs/by-name/me/meteo-qt/package.nix:11]:
```suggestion
  format = "setuptools";

```

Comment (Aleksanaa) [reply pkgs/by-name/me/meteo-qt/package.nix:11]: Not needed

File: pkgs/by-name/me/meteo-qt/package.nix [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  dependencies = with python3Packages; [
+    lxml
+    pyqt5
+  ];
+
+  pythonImportsCheck = [ "meteo_qt" ];
+
+  makeWrapperArgs = [ "\${qtWrapperArgs[@]}" ];
+
+  meta = {
```
Comment (FliegendeWurst) [inline pkgs/by-name/me/meteo-qt/package.nix:36]: Please set either platforms = linux or platforms = unix, depending on where the program can run.

**Review by FliegendeWurst — APPROVED** [review]

**Comment by FliegendeWurst** [comment]
By the way, you may want to add your git email to Github so it show your commits correctly.

**Comment by FliegendeWurst** [comment]
The commit would not have your profile picture next to it.

**Pull request merged**